diff --git a/src/dtx/tests/dts_utils.c b/src/dtx/tests/dts_utils.c
index c884872143..607b20d732 100644
--- a/src/dtx/tests/dts_utils.c
+++ b/src/dtx/tests/dts_utils.c
@@ -147,7 +147,7 @@ setup_local_args(void **state)
 	int                    int_flag;
 	int                    rc;
 
-	test_args_init(&test_args, VPOOL_SIZE);
+	test_args_init(&test_args, VPOOL_64M);
 
 	int_flag = is_daos_obj_type_set(arg->otype, DAOS_OT_DKEY_UINT64);
 
diff --git a/src/vos/tests/vts_common.h b/src/vos/tests/vts_common.h
index b7dc09dcfe..0e3c7fa3a9 100644
--- a/src/vos/tests/vts_common.h
+++ b/src/vos/tests/vts_common.h
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright 2016-2023 Intel Corporation.
+ * (C) Copyright 2016-2024 Intel Corporation.
  *
  * SPDX-License-Identifier: BSD-2-Clause-Patent
  */
@@ -35,6 +35,7 @@
 	} while (0)
 #endif /* FAULT_INJECTION */
 
+#define VPOOL_64M        (64ULL << 20)
 #define VPOOL_256M	(256ULL << 20)
 #define VPOOL_1G	(1ULL << 30)
 #define VPOOL_2G	(2ULL << 30)
