diff --git a/src/vos/vos_common.c b/src/vos/vos_common.c
index 0e274437a3..a5fce7d36a 100644
--- a/src/vos/vos_common.c
+++ b/src/vos/vos_common.c
@@ -1028,10 +1028,10 @@ vos_self_init_ext(const char *db_path, bool use_sys_db, int tgt_id, bool nvme_in
 	if (rc)
 		goto failed;
 
-	if (use_sys_db)
-		rc = vos_db_init(db_path);
-	else
-		rc = vos_db_init_ex(db_path, "self_db", true, true);
+	// if (use_sys_db)
+	rc = vos_db_init(db_path);
+	// else
+	// 	rc = vos_db_init_ex(db_path, "self_db", true, true);
 	if (rc)
 		goto failed;
 
