diff --git a/src/dtx/tests/dtx_tests.c b/src/dtx/tests/dtx_tests.c
index b06a2f4ea8..7236a2bda0 100644
--- a/src/dtx/tests/dtx_tests.c
+++ b/src/dtx/tests/dtx_tests.c
@@ -61,6 +61,7 @@ int
 main(int argc, char **argv)
 {
 	int                  rc        = 0;
+	char                 filter[1024];
 	int                  nr_failed = 0;
 	int                  opt       = 0;
 	int                  index     = 0;
@@ -107,8 +108,6 @@ main(int argc, char **argv)
 #if CMOCKA_FILTER_SUPPORTED == 1 /** requires cmocka 1.1.5 */
 		{
 			/** Add wildcards for easier filtering */
-			char filter[sizeof(optarg) + 2];
-
 			sprintf(filter, "*%s*", optarg);
 			cmocka_set_test_filter(filter);
 			printf("Test filter: %s\n", filter);
