From d633d3b0a5f03be280efb80a69b9d5ed4e9c4d56 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C5=81ukasz=20Stolarczuk?= <lukasz.stolarczuk@intel.com>
Date: Tue, 14 Jul 2020 13:58:34 +0200
Subject: [PATCH] fix generating gcov files and turn-off verbose log

---
 codecov | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/codecov b/codecov
index e702ecd..0a2f4d8 100755
--- a/codecov
+++ b/codecov
@@ -1108,9 +1108,9 @@ then
     if [ "$ft_gcovout" = "0" ];
     then
       # suppress gcov output
-      bash -c "find $proj_root -type f -name '*.gcno' $gcov_include $gcov_ignore -exec $gcov_exe -pb $gcov_arg {} +" >/dev/null 2>&1 || true
+      bash -c "find $proj_root -type f -name '*.gcno' $gcov_include $gcov_ignore -execdir $gcov_exe -pb $gcov_arg {} \;" >/dev/null 2>&1 || true
     else
-      bash -c "find $proj_root -type f -name '*.gcno' $gcov_include $gcov_ignore -exec $gcov_exe -pb $gcov_arg {} +" || true
+      bash -c "find $proj_root -type f -name '*.gcno' $gcov_include $gcov_ignore -execdir $gcov_exe -pb $gcov_arg {} \;" || true
     fi
   else
     say "${e}==>${x} gcov disabled"
@@ -1425,7 +1425,7 @@ do
       report_len=$(wc -c < "$file")
       if [ "$report_len" -ne 0 ];
       then
-        say "    ${g}+${x} $file ${e}bytes=$(echo "$report_len" | tr -d ' ')${x}"
+        #say "    ${g}+${x} $file ${e}bytes=$(echo "$report_len" | tr -d ' ')${x}"
         # append to to upload
         _filename=$(basename "$file")
         if [ "${_filename##*.}" = 'gcov' ];
-- 
2.25.1

