From b5179dc4822eaab192361da05aa95d98f523960f Mon Sep 17 00:00:00 2001
From: Lukasz Dorau <lukasz.dorau@intel.com>
Date: Mon, 7 May 2018 12:05:40 +0200
Subject: [PATCH] travis: fix travisci_build_coverity_scan.sh

---
 travisci_build_coverity_scan.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/travisci_build_coverity_scan.sh b/travisci_build_coverity_scan.sh
index ad9d4afcf..562b08bcc 100644
--- a/travisci_build_coverity_scan.sh
+++ b/travisci_build_coverity_scan.sh
@@ -92,8 +92,8 @@ response=$(curl \
   --form description="Travis CI build" \
   $UPLOAD_URL)
 status_code=$(echo "$response" | sed -n '$p')
-if [ "$status_code" != "201" ]; then
+if [ "$status_code" != "200" ]; then
   TEXT=$(echo "$response" | sed '$d')
-  echo -e "\033[33;1mCoverity Scan upload failed: $TEXT.\033[0m"
+  echo -e "\033[33;1mCoverity Scan upload failed: $response.\033[0m"
   exit 1
 fi
-- 
2.13.6

